lib/ansible/galaxy/collection.py

```diff
@@ -1008,6 +1064,47 @@ def reset_stat(tarinfo):
         display.display('Created collection for %s at %s' % (collection_name, to_text(b_tar_path)))
 
 
+def _build_collection_dir(b_collection_path, b_collection_output, collection_manifest, file_manifest):
+    """ Build a collection directory from the manifest data. This should follow the same pattern as _build_collection_tar. """
+    os.makedirs(b_collection_output, mode=0o0755)
+
+    files_manifest_json = to_bytes(json.dumps(file_manifest, indent=True), errors='surrogate_or_strict')
+    collection_manifest['file_manifest_file']['chksum_sha256'] = secure_hash_s(files_manifest_json, hash_func=sha256)
+    collection_manifest_json = to_bytes(json.dumps(collection_manifest, indent=True), errors='surrogate_or_strict')
+
+    # Write contents to the files
+    for name, b in [('MANIFEST.json', collection_manifest_json), ('FILES.json', files_manifest_json)]:
+        b_path = os.path.join(b_collection_output, to_bytes(name, errors='surrogate_or_strict'))
+        b_io = BytesIO(b)
+        with open(b_path, 'wb') as file_obj:
+            shutil.copyfileobj(b_io, file_obj)
+
+        os.chmod(b_path, 0o0644)
+
+    base_directories = []
+    for file_info in file_manifest['files']:
+        if file_info['name'] == '.':
+            continue
+
+        src_file = os.path.join(b_collection_path, to_bytes(file_info['name']))
+        dest_file = os.path.join(b_collection_output, to_bytes(file_info['name']))
+
+        if any([src_file.startswith(directory) for directory in base_directories]):
```

Comment:
Creating in-memory throw-away lists can be avoided when using functions that support iterables. Besides saving some memory, it's also faster because the generator is not necessarily looped over completely:
```suggestion
        if any(src_file.startswith(directory) for directory in base_directories):
```